packages/flutter_tools/test/integration.shard/android_plugin_example_app_build_test.dart

```diff
@@ -36,114 +33,28 @@ void main() {
     ], workingDirectory: tempDir.path);
-    if (result.exitCode != 0) {
-      throw Exception('flutter create failed: ${result.exitCode}\n${result.stderr}\n${result.stdout}');
-    }
-
-    final Directory exampleAppDir =
-        tempDir.childDirectory(testName).childDirectory('example');
-
-    final File buildGradleFile = exampleAppDir.childDirectory('android').childFile('build.gradle.kts');
-    expect(buildGradleFile, exists);
-
-    final String buildGradle = buildGradleFile.readAsStringSync();
-    final RegExp androidPluginRegExp =
-        RegExp(r'com\.android\.tools\.build:gradle:(\d+\.\d+\.\d+)');
-
-    // Use AGP 7.2.0
-    final String newBuildGradle = buildGradle.replaceAll(
-        androidPluginRegExp, 'com.android.tools.build:gradle:7.2.0');
-    buildGradleFile.writeAsStringSync(newBuildGradle);
-
-    // Run flutter build apk using AGP 7.2.0
-    result = processManager.runSync(<String>[
-      flutterBin,
-      ...getLocalEngineArguments(),
-      'build',
-      'apk',
-      '--target-platform=android-arm',
-    ], workingDirectory: exampleAppDir.path);
-    if (result.exitCode != 0) {
-      throw Exception('flutter build failed: ${result.exitCode}\n${result.stderr}\n${result.stdout}');
-    }
-
-    final File exampleApk = fileSystem.file(fileSystem.path.join(
-      exampleAppDir.path,
-      'build',
-      'app',
-      'outputs',
-      'flutter-apk',
-      'app-release.apk',
-    ));
-    expect(exampleApk, exists);
-
-    if (template == 'plugin_ffi') {
-      // Does not support AGP 3.3.0.
-      return;
-    }
-
-    // Clean
-    result = processManager.runSync(<String>[
-      flutterBin,
-      ...getLocalEngineArguments(),
-      'clean',
-    ], workingDirectory: exampleAppDir.path);
-    if (result.exitCode != 0) {
-      throw Exception('flutter clean failed: ${result.exitCode}\n${result.stderr}\n${result.stdout}');
-    }
-
-    // Remove Gradle wrapper
-    fileSystem
-        .directory(fileSystem.path
-            .join(exampleAppDir.path, 'android', 'gradle', 'wrapper'))
-        .deleteSync(recursive: true);
-
-    // Enable R8 in gradle.properties
-    final File gradleProperties =
-        exampleAppDir.childDirectory('android').childFile('gradle.properties');
-    expect(gradleProperties, exists);
-
-    gradleProperties.writeAsStringSync('''
-org.gradle.jvmargs=-Xmx4G -XX:MaxMetaspaceSize=2G -XX:+HeapDumpOnOutOfMemoryError
-android.useAndroidX=true
-android.enableJetifier=true
-android.enableR8=true''');
-
-    // Run flutter build apk using AGP 3.3.0
-    result = processManager.runSync(<String>[
-      flutterBin,
-      ...getLocalEngineArguments(),
-      'build',
-      'apk',
-      '--target-platform=android-arm',
-    ], workingDirectory: exampleAppDir.path);
-    if (result.exitCode != 0) {
-      throw Exception('flutter build failed: ${result.exitCode}\n${result.stderr}\n${result.stdout}');
-    }
-    expect(exampleApk, exists);
+    expect(result, const ProcessResultMatcher());
```

Comment:
This would no longer be a test of building the example app if we remove that part, so I think we should either retain it or change the name

Reply:
Oh oops I cut out too much, re-adding the example app.